python/src/dataset.rs

```diff
@@ -71,6 +71,7 @@ impl Dataset {
             block_size,
             index_cache_size: index_cache_size.unwrap_or(DEFAULT_INDEX_CACHE_SIZE),
             session: None,
+            os_options: None,
```

Comment:
nit: if we can call this `store_options` or `object_store_options`, I'd prefer that. OS to me reads as "operating system".